pkg/ddevapp/drupal.go

```diff
@@ -182,79 +210,153 @@ $databases['default']['default'] = array(
+		output.UserOut.Warnf("Failed to write .gitignore: %v", err)
 	}
 
-	return settingsFilePath, nil
+	return app.SiteLocalSettingsPath, nil
 }
 
-// createDrupal6SettingsFile creates the app's settings.php or equivalent,
-// adding things like database host, name, and password
-// Returns the fullpath to settings file and err
-func createDrupal6SettingsFile(app *DdevApp) (string, error) {
+// createDrupal8SettingsFile manages creation and modification of settings.php and settings.ddev.php.
+// If a settings.php file already exists, it will be modified to ensure that it includes
+// settings.ddev.php, which contains ddev-specific configuration.
+func createDrupal8SettingsFile(app *DdevApp) (string, error) {
+	// Currently there isn't any customization done for the drupal config, but
+	// we may want to do some kind of customization in the future.
+	drupalConfig := NewDrupalSettings()
 
-	settingsFilePath, err := app.DetermineSettingsPathLocation()
-	if err != nil {
-		return "", fmt.Errorf("Failed to get Drupal 6 settings file path: %v", err.Error())
+	if err := manageDrupalCommonSettingsFile(app, drupalConfig); err != nil {
+		return "", err
+	}
+
+	// Create settings.ddev.php if one doesn't already exist
+	if fileutil.FileExists(app.SiteLocalSettingsPath) {
+		output.UserOut.Printf("ddev settings file %s already exists", drupalConfig.SiteSettingsLocal)
+	} else {
+		output.UserOut.Printf("No %s exists, creating one", drupalConfig.SiteSettingsLocal)
+		if err := writeDrupal8SettingsFile(drupalConfig, app.SiteLocalSettingsPath); err != nil {
+			return "", fmt.Errorf("failed to write Drupal settings file: %v", err.Error())
+		}
 	}
-	output.UserOut.Printf("Generating %s file for database connection.", filepath.Base(settingsFilePath))
 
+	if err := createGitIgnore(filepath.Dir(app.SiteSettingsPath), drupalConfig.SiteSettingsLocal); err != nil {
+		output.UserOut.Warnf("Failed to write .gitignore: %v", err)
+	}
+
+	return app.SiteLocalSettingsPath, nil
+}
+
+// createDrupal6SettingsFile manages creation and modification of settings.php and settings.ddev.php.
+// If a settings.php file already exists, it will be modified to ensure that it includes
+// settings.ddev.php, which contains ddev-specific configuration.
+func createDrupal6SettingsFile(app *DdevApp) (string, error) {
 	// Currently there isn't any customization done for the drupal config, but
 	// we may want to do some kind of customization in the future.
 	drupalConfig := NewDrupalSettings()
 
-	err = writeDrupal6SettingsFile(drupalConfig, settingsFilePath)
-	if err != nil {
-		return settingsFilePath, fmt.Errorf("Failed to write Drupal settings file: %v", err.Error())
+	if err := manageDrupalCommonSettingsFile(app, drupalConfig); err != nil {
+		return "", err
 	}
 
-	return settingsFilePath, nil
+	// Create settings.ddev.php if one doesn't already exist
+	if fileutil.FileExists(app.SiteLocalSettingsPath) {
```

Comment:
Eventually we'll hate the repeated nature of these things and the fact that we have to maintain the same thing in multiple places  that we'll consolidate with functions (or somehow).  Eventually.